unlock-app/src/components/creator/CreatorLockForm.js

```diff
@@ -29,59 +29,81 @@ import {
 
 import { INFINITY, UNLIMITED_KEYS_COUNT } from '../../constants'
 
+/**
+ * Converts the lock values into form values
+ * @param {*} lockValues
+ */
+export const lockToFormValues = lockValues => {
+  const formValues = Object.assign({}, lockValues)
+
+  // In the form, duration is shown in days, vs seconds in the lock object
+  formValues.expirationDuration =
+    lockValues.expirationDuration / lockValues.expirationDurationUnit
+
+  // Unlimited keys
+  if (lockValues.maxNumberOfKeys === UNLIMITED_KEYS_COUNT) {
```

Comment:
Chris added `unlimitedKeys` to locks in the web3Service, so you can use that directly as in:

```js
  if (lockValues.unlimitedKeys) {
```

Reply:
So... I am not sure this move was right :/ We discussed (very vocally if you recall on @cnasc 's first day) that the services should not be defining the "format" of the data being exchanged in the application... the `lock` propTypes as defined in `propTypes.js` does not have `unlimitedKeys`.

Reply:
To rephrase this I think it is not right that code used inside of the react components has a format that is defined by the web3Service.

Reply:
Oh! My mistake if I misunderstood, I was working with Chris to do something similar to this:

```js
    const attributes = {
      keyPrice: x => Web3Utils.fromWei(x, 'ether'),
      expirationDuration: parseInt,
      maxNumberOfKeys: value => {
        if (value === MAX_UINT) {
          return UNLIMITED_KEYS_COUNT
        }
        return parseInt(value)
      },
      owner: x => x,
      outstandingKeys: parseInt,
    }
```

Which was a change to convert keyPrice to eth, as the standard representation of keyPrice throughout unlock-app.

The idea to represent unlimited keys as the boolean `unlimitedKeys` was a similar rationale, which was to say "inside unlock-app, unlimited keys will be represented by this bool" and then to convert to and from the ethereum-specific representation of it inside the web3Service.

I see the missed spot in propTypes, it should definitely be an issue opened to fix if once (and if) we agree that this idea is consistent.

The original issue, for reference is in this PR: #1526 and you can see in my comment the source of my (potential) misunderstanding:

> I would like to take the same approach Julien used for keyPrice, which is to do the conversion in the walletService, and use the same representation throughout the frontend. This is probably more lines of code to touch (especially in the tests) but is a cleaner way to do it, and will ultimately make it a lot easier to move to hooks too. If you want to chat later this afternoon about this, I am around

The final PR was #1543